src/host/CursorBlinker.cpp

```diff
@@ -82,8 +82,10 @@ void CursorBlinker::TimerRoutine(SCREEN_INFORMATION& ScreenInfo) const noexcept
     auto& cursor = buffer.GetCursor();
     auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
     auto* const pAccessibilityNotifier = ServiceLocator::LocateAccessibilityNotifier();
+    const bool inConpty{ gci.IsInVtIoMode() };
 
-    if (!WI_IsFlagSet(gci.Flags, CONSOLE_HAS_FOCUS))
+    // GH#2988: ConPTY can now be focused, but it doesn't need to do any of this work either.
+    if (inConpty || !WI_IsFlagSet(gci.Flags, CONSOLE_HAS_FOCUS))

```

Comment:
Plz file a followup to make sure that this timer doesn't even get scheduled? For every hidden OpenConsole window, we are burning a timer every 250-500ms to... check some booleans that will never be true.

Reply:
* `CONSOLE_INFORMATION` creates `_blinker` as a stack member
* `CursorBlinker` ctor does `CreateThreadpoolTimer`
* We don't start the timer till the call to `SetCaretTimer`
* `SetCaretTimer` is called in two places:
  * `CursorBlinker::FocusStart`, which is only called in `WM_SETFOCUS` in `wndproc.cpp`, so that's never hit for conpty
  * `CursorBlinker::SettingsChanged`, which is conveniently only called `WM_SETTINGCHANGE`. So also never for conpty.
  
It's probably coincidence that the blinker was never started for conpty, but I'm not terribly afraid of this regressing any time soon